src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -2854,7 +2854,7 @@ QueryTreeNodePtr QueryAnalyzer::tryResolveIdentifierFromAliases(const Identifier
     {
         const auto top_expression = scope.expressions_in_resolve_process_stack.getTop();
 
-        if (!isNodePartOfTree(top_expression.get(), root_expression_with_alias.get()))
+        if (identifier_lookup.isExpressionLookup() && !isNodePartOfTree(top_expression.get(), root_expression_with_alias.get()))
```

Comment:
Why do we even get here? In that case we should have used `scope.alias_name_to_lambda_node` and there should be nothing by name `path`

Reply:
https://github.com/ClickHouse/ClickHouse/blob/cf148865862e0eda2533226e232b69da67a76225/src/Analyzer/Passes/QueryAnalysisPass.cpp#L1043-L1045

I don't really know, but, probably, from the context `x AS path` we can't tell whether `x` is an expression or function (except in the place where we collect aliases, where nothing is resolved).